subsys/net/lib/sockets/sockets_inet.c

```diff
@@ -664,6 +680,11 @@ ssize_t zsock_sendmsg_ctx(struct net_context *ctx, const struct msghdr *msg,
 	k_timepoint_t buf_timeout, end;
 	int status;
 
+	if (net_context_get_type(ctx) == SOCK_RAW) {
```

Comment:
Why do we prevent `sendmsg` for raw sockets?

Reply:
Probably no good reason now that I think of it, I can drop this restriction and add some tests for sendmsg()/recvmsg().